frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -62,19 +62,39 @@ pub struct ConstMetadataDef {
 	pub doc: Vec<syn::Lit>,
 }
 
-impl syn::parse::Parse for ConstMetadataDef  {
-	fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
-		let doc = helper::get_doc_literals(&syn::Attribute::parse_outer(input)?);
-		input.parse::<syn::Token![type]>()?;
-		let ident = input.parse::<syn::Ident>()?;
-		input.parse::<syn::Token![:]>()?;
-		input.parse::<keyword::Get>()?;
-		input.parse::<syn::Token![<]>()?;
-		let mut type_ = input.parse::<syn::Type>()?;
-		type_ = syn::parse2::<syn::Type>(replace_self_by_t(type_.to_token_stream()))
+impl TryFrom<&syn::TraitItemType> for ConstMetadataDef {
+	type Error = syn::Error;
+
+	fn try_from(trait_ty: &syn::TraitItemType) -> Result<Self, Self::Error> {
+		let doc = helper::get_doc_literals(&trait_ty.attrs);
+		let ident = trait_ty.ident.clone();
+		let bound = trait_ty.bounds
+			.iter()
+			.find_map(|b|
+				if let syn::TypeParamBound::Trait(tb) = b {
+					tb.path.segments
+						.last()
+						.and_then(|s| if s.ident == "Get" { Some(s) } else { None } )
+				} else {
+					None
+				}
+			)
+			.ok_or_else(|| Error::new(trait_ty.span(), "`Get<T>` trait bound not found"))?;
```

Comment:
I think for all error message you can add that this is due to `#[pallet::constant]` constraint,
Something like "Invalid usage of `#[pallet::constant], "